### PR TITLE
Refactored base-email-view

### PR DESCRIPTION
**Pull request by MihirKohli**
Resolved https://github.com/openwisp/openwisp-users/issues/354

### CODE REVIEW

File: openwisp_users/api/views.py [hunk 1]
```diff
@@ -174,8 +176,8 @@ def get_object(self):
         user = User.objects.filter(id=self.request.user.id)
         qs = self.get_queryset()
         if (
-            user.first().is_staff is True
-            and not qs.filter(pk=self.request.user.id).exists()
+                user.first().is_staff is True
+                and not qs.filter(pk=self.request.user.id).exists()
```
Comment (nemesifier) [inline openwisp_users/api/views.py:180]: this should not be changed

File: openwisp_users/api/views.py [hunk 2]
```diff
@@ -198,44 +200,15 @@ def update(self, request, *args, **kwargs):
...
-        for org_user in org_users:
-            if org_user.is_admin:
-                qs_user = qs_user | org_user.organization.users.all().distinct()
-        qs_user = qs_user.filter(is_superuser=False)
-        return qs_user.filter(pk=self.kwargs['pk'])
-
     def get_serializer_context(self):
         if getattr(self, 'swagger_fake_view', False):
-            # To get rid of assertion error raised in
-            # the dev server, and for schema generation
```
Comment (nemesifier) [inline openwisp_users/api/views.py:238]: do not remove these comments please

**Review by nemesifier — CHANGES_REQUESTED** [review]
@MihirKohli please check the details of the build / QA check failures.

File: openwisp_users/api/views.py (hunk 2 above)
Comment (MihirKohli) [reply openwisp_users/api/views.py]: > do not remove these comments please

updated commits review it please

**Comment by MihirKohli** [comment]
made changes do check it

File: openwisp_users/api/views.py [hunk 3]
```diff
@@ -198,7 +199,7 @@ def update(self, request, *args, **kwargs):
         )
 
 
-class BaseEmailView(ProtectedAPIMixin, GenericAPIView):
+class BaseEmailView(ProtectedAPIMixin, FilterByParent, GenericAPIView):
```
Comment (nemesifier) [inline openwisp_users/api/views.py:202]: Yeah ok, we're inheriting from this `FilterByParent` now but there's some code here in BaseEmailView which is now redundant and has to be removed, which is not being done in this patch, so the goal of the issue is not being reached yet.

**Review by nemesifier — CHANGES_REQUESTED** [review]
> made changes do check it

Please change the commit message to:

```
[refactor] Refactor BaseEmailView to use the FilterByParent mixin #354

Closes #354
```

**Pull request closed without merging**